Assets/_Functionalities/Wms/Scripts/WMSImportAdapter.cs

```diff
@@ -0,0 +1,352 @@
+        {
+            string value = string.Empty;
+            string p = param + "=";
+            if (sourceUrl.ToLower().Contains(p))
+            {
+                value = sourceUrl.ToLower().Split(p)[1].Split("&")[0];
+            }
+            return value;
+        }
+
+        private void AddWMSLayer(WMS.WMSLayerQueryParams layer, string sourceUrl, FolderLayer folderLayer, bool defaultEnabled)
+        {
+            //Spawn a new WMS layer
+            WMSLayerGameObject newLayer = Instantiate(layerPrefab);
+            newLayer.LayerData.SetParent(folderLayer);
+            newLayer.Name = layer.name;            
+            UriBuilder uriBuilder = CreateLayerUri(layer, sourceUrl);
+            var uri = uriBuilder.Uri.ToString();
+            newLayer.SetUrlPropertyData(uri);
+            newLayer.LayerData.ActiveSelf = defaultEnabled;
+        }
+
+        private UriBuilder CreateLayerUri(WMS.WMSLayerQueryParams layer, string sourceUrl)
+        {
+            // Start by removing any query parameters we want to inject
+            var uriBuilder = new UriBuilder(sourceUrl);
+
+            // Make sure we have a wmsVersion set
+            if (string.IsNullOrEmpty(wmsVersion))
+            {
+                Debug.LogWarning("WMS version could not be determined, defaulting to " + defaultFallbackVersion);
+                wmsVersion = defaultFallbackVersion;
+            }
+
+            var parameters = new NameValueCollection();
+            uriBuilder.TryParseQueryString(parameters);
+
+            string encoded = Uri.EscapeDataString(layer.name);
+
+            // Set the required query parameters for the GetMap request
+            uriBuilder.SetQueryParameter("service", "WMS");
+            uriBuilder.SetQueryParameter("version", wmsVersion);
+            uriBuilder.SetQueryParameter("request", "GetMap");
+
+            uriBuilder.AddQueryParameter("layers", encoded);
+            uriBuilder.AddQueryParameter("styles", layer.style);
+            uriBuilder.AddQueryParameter(layer.spatialReferenceType, layer.spatialReference);
+            uriBuilder.AddQueryParameter("bbox", "{0}"); // Bbox value is injected by ImageProjectionLayer
+            uriBuilder.AddQueryParameter("width", layerPrefab.PreferredImageSize.x.ToString());
+            uriBuilder.AddQueryParameter("height", layerPrefab.PreferredImageSize.y.ToString());
+            string format = GetParamValueFromSourceUrl(sourceUrl, "format");
+            format = Uri.UnescapeDataString(format);
+            if (format != "image/png" && format != "image/jpeg")
+                format = "image/png";
+            uriBuilder.AddQueryParameter("format", format);
+            if (!sourceUrl.Contains("transparent="))
+                uriBuilder.AddQueryParameter("transparent", layerPrefab.TransparencyEnabled.ToString());
+            return uriBuilder;
+        }
+
+        private class WMS
```

Comment:
Kan deze class voor het overzicht in een aparte file?

Reply:
heb de structuur van de wfsimportadapter aangehouden. nu zijn ze consistent. 
ik ben het met je eens en zou dit liever ook zien. Maar denk dat dit  een refactor slag moet worden waarin de wfs importadapter meegenomen moet worden. Ook geld voor beiden dat de xml.serializer ook dan meteen kan worden meegenomen